Framework/Crystal/src/IndexPeaks.cpp

```diff
@@ -44,11 +55,27 @@ void IndexPeaks::init() {
                             "AverageError", 0.0, Direction::Output),
                         "Gets set with the average HKL indexing error.");
 
-  this->declareProperty("RoundHKLs", true,
-                        "Round H, K and L values to integers");
+  this->declareProperty(make_unique<PropertyWithValue<int>>(
+                            "TotalNumIndexed", 0, Direction::Output),
+                        "Gets set with the number of Total indexed peaks.");
 
-  this->declareProperty("CommonUBForAll", false,
-                        "Index all orientations with a common UB");
+  this->declareProperty(make_unique<PropertyWithValue<int>>("MainNumIndexed", 0,
+                                                            Direction::Output),
+                        "Gets set with the number of indexed main peaks.");
+
+  this->declareProperty(make_unique<PropertyWithValue<int>>("SateNumIndexed", 0,
+                                                            Direction::Output),
+                        "Gets set with the number of indexed main peaks.");
+
+  this->declareProperty(
+      make_unique<PropertyWithValue<double>>("MainError", 0.0,
+                                             Direction::Output),
+      "Gets set with the average HKL indexing error of Main Peaks.");
+
+  this->declareProperty(
+      make_unique<PropertyWithValue<double>>("SatelliteError", 0.0,
+                                             Direction::Output),
+      "Gets set with the average HKL indexing error of Satellite Peaks.");
```

Comment:
There's no testing for these new parameters? How do we know if it works? How do we know when it breaks?